applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,497 @@
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template< class TElementData >
+void DVMSDEMCoupled<TElementData>::AddMassStabilization(
+    TElementData& rData,
+    MatrixType &rMassMatrix)
+{
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const array_1d<double,3> convective_velocity = this->FullConvectiveVelocity(rData);
+
+    double tau_one;
+    double tau_two;
+    double tau_p;
+    this->CalculateStabilizationParameters(rData,convective_velocity,tau_one,tau_two,tau_p);
+
+    const double dt = rData.DeltaTime;
+
+    Vector AGradN;
+    this->ConvectionOperator(AGradN,convective_velocity,rData.DN_DX);
+
+    // Multiplying convective operator by density to have correct units
+    AGradN *= density;
+
+    const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+
+    double W = rData.Weight * tau_one * density; // This density is for the dynamic term in the residual (rho*Du/Dt)
+
+    // Note: Dof order is (u,v,[w,]p) for each node
+    for (unsigned int i = 0; i < NumNodes; i++) {
+        unsigned int row = i*BlockSize;
+
+        for (unsigned int j = 0; j < NumNodes; j++) {
+            unsigned int col = j*BlockSize;
+
+            // u*grad(v) * TauOne * du/dt
+            // v * TauOne/dt * du/dt (from v*d(uss)/dt)
+            double K = W * (AGradN[i] - rData.N[i]/dt) * rData.N[j];
+
+            for (unsigned int d = 0; d < Dim; d++)
+            {
+                rMassMatrix(row+d,col+d) += K;
+                // grad(q) * TauOne * du/dt
+                double UGAlpha = W * fluid_fraction * rData.DN_DX(i,d) * rData.N[j];
+                rMassMatrix(row+Dim,col+d) += UGAlpha;
+            }
+        }
+    }
+}
+
+template< class TElementData >
+void DVMSDEMCoupled<TElementData>::CalculateStabilizationParameters(
+    const TElementData& rData,
+    const array_1d<double,3> &Velocity,
+    double &TauOne,
+    double &TauTwo,
+    double &TauP) const
+{
+    const double h = rData.ElementSize;
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const double viscosity = this->GetAtCoordinate(rData.EffectiveViscosity,rData.N);
+    constexpr double mTauC1 = DVMS<TElementData>::mTauC1;
+    constexpr double mTauC2 = DVMS<TElementData>::mTauC2;
+
+    double velocity_norm = Velocity[0]*Velocity[0];
+    for (unsigned int d = 1; d < Dim; d++)
+        velocity_norm += Velocity[d]*Velocity[d];
+    velocity_norm = std::sqrt(velocity_norm);
+
+    double inv_tau = mTauC1 * viscosity / (h*h) + density * ( 1.0/rData.DeltaTime + mTauC2 * velocity_norm / h );
+    TauOne = 1.0/inv_tau;
+    TauTwo = viscosity + density * mTauC2 * velocity_norm * h / mTauC1;
+
+    // Auxiliary coefficient StaticTauOne*TauTwo/Dt that appears on the pressure subscale model
+    // This coefficient comes from mass equation: div(u^n+1 + u^n) = 0 setting to 0 is to consider
+    // div(u^n+1) = 0
+    TauP = 0.0;
```

Comment:
I would remove all reference to this term. If we want, we can add it later, but it is better to leave everything clean

Reply:
Indeed, aren't these the standard tau constant used in the base NS formulation? If so it is the base class method could be used.